Drobes

Batrachophreno opened the pull request:
[Original suggestion here.](https://forums.aurorastation.org/topic/22454-tidy-up-lockers/)
- Replaces departmental/job-specific wardrobe lockers with new 'Drobe' vending machines, each containing pertinent clothing, all with new icons: ClothesMate, EngiDrobe, AtmosDrobe, HyDrobe, SecDrobe, BarDrobe, ChefDrobe, ChapelDrobe, SciDrobe, MediDrobe, ChemDrobe, CargoDrobe, RoboDrobe
- Makes vending machine ads populate at the top of the purchasing UI and randomly cycle.
- Few vending-machine related map touch-ups (mostly vending machines missing the floor 'hatched' pattern underneath).
- Misc code readability improvements.

Greenjoe12345 commented on code/game/machinery/vending/_vending.dm:
```diff
@@ -161,24 +186,61 @@
+	/**
+	  * Is this item on station or not
+	  *
+	  * if it doesn't originate from off-station during mapload, all_products_free gets automatically set to TRUE if it was unset previously.
```
wording here is a bit funny, also thinking
places like port of call maps make sense to still have vending machines that cost money, those would use an override?

Batrachophreno commented:
!review

alsoandanswer commented:
Hygienevendors and Toxicmates shouldn't be categorized as Horizon vendors - former is generic, latter seems to be a really old science vendor when they made bombs on the regular

Batrachophreno commented:
!wip

Batrachophreno commented:
Responses:
**Botanical gloves:** Good to know, removed!
**Hydroponics closet count:** Fair enough, restored the second closet and tried to squeeze the vendor in nearby. You know the tyranny of cramped spaces.
**Raskariim mask:** Damn alright, I didn't recognize that was for 'baddies' on Horizon- xenolore weakness. Moved it to the contraband section.
**Recolorable items:** Also fair enough- once TGUI is updated we'll be able to port over TGcode that lets you recolor directly from vending machines, so I just removed the white ones for now.
**Lavatory Essentials and Toximate 3000:** They're really only in 'misc_horizon.dm' because they're Miscellaneous vending machines that AREN'T strictly Offship or Away Site machines. The latter I also know is legacy, but I saw no reason to remove it. I'll take name recommendations for the file if 'misc_horizon' doesn't communicate that those are Miscellaneous machines that aren't restricted to offships or away sites! I just didn't want to make four more files each with a single vending machine in them.

Batrachophreno commented:
!review

alsoandanswer commented:
misc_horizon implies that it's horizon only, maybe just name the file "miscellaneous"

i think the lavatory-vend fits in service well enough, the toxicmate i don't really care for

hazelrat commented:
Good to approve once the conflicts are cleared!

Batrachophreno commented:
!review
